src/data/docs/abilities.md

```diff
@@ -0,0 +1,562 @@
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      greet : Text ->{IO} ()
+      msg   : Text -> Text
+   
+  Now evaluating any watch expressions (lines starting with
+  `>`)... Ctrl+C cancels.
+
+```
+Notice the `{IO}` attached to the `->` in `greet : Text ->{IO} ()`. We read this type signature as saying that `greet` is a function from `Text` to `()` which "does some I/O in the process". `IO` is what we call an _ability_ in Unison and we say that `greet` "requires the `IO` ability".
+
+This tutorial covers the basics of how you'll interact with abilities and how they are typechecked, using `IO` as an example, then shows how you can create and use new abilities. Abilities are a simple-to-use but powerful language feature that subsumes many other more specific language features: exceptions, asynchronous I/O, generators, coroutines, logic programming, and many more concepts can all just be expressed as regular libraries in Unison, using abilities.
+
+Let's get started! This tutorial has some (optional, extremely fun) exercises, with answers at the end.
+
+> 📚  Unison's "abilities" are called _algebraic effects_ in the literature. See the [bibliography](/docs/bibliography/#programming-language-theory) if you're interested in the research behind this aspect of Unison.
+>
+> Also see [the language reference section on abilities](/docs/language-reference#abilities).
+
+<a id="usage"></a>
+
+## Usage of abilities and ability checking
+
+As we've seen, ability requirements are part of Unison's type system. Functions like `greet : Text ->{IO} ()` that do `IO` (like `printLine`, which prints a line to the console) indicate this fact in their type signature.
+
+A function can have 0 or more abilities inside the `{}`, separated by commas if there's more than one, and a function that requires no abilities is sometimes called _pure_. If we try to give `greet` a type signature that says it's pure, we'll get a type error called an _ability check failure_ because the function's implementation still requires `IO` but its type doesn't make that ability available:
+
+```unison
+greet : Text ->{} ()
+greet name =
+  printLine ("Hello there, " ++ name)
+```
+
+```ucm
+
+  The expression in red needs the {base.io.IO} ability, but this location does not have access to any abilities.
+  
+      3 |   printLine ("Hello there, " ++ name)
+  
+
+```
+The empty `{}` attached to the arrow on the `greet` type signature tells the typechecker we are expecting `greet` to be pure. The typechecker therefore complains when the function tries to call a function (`printLine`) that requires abilities. _The ability requirements of a function `f` must include all the abilities required by any function that could be called in the body of `f`._
+
+> 😲  Pretty nifty! We can constrain what sorts of operations a function can access just by giving it a different type signature.
```

Comment:
Perhaps more importantly a caller can determine which operations a function can access just by looking at the type signature. They can call the function without worrying about unexpected side effects (will this call throw an exception? modify state? launch the nukes?!?).

It's the same selling point, really, but maybe shifting the focus like this will make it resonate more?